### ISSUE TITLE
Major issue with language filter!  Messages are still sent even if they are cancelled, respond ASAP!

### ISSUE THREAD

**Issue by c0wg0d** [kangarko/ChatControl-Pro#623]
BungeeCord version: This server is running BungeeCord version git:BungeeCord-Bootstrap:1.12-SNAPSHOT:e23195f:1299 by md_5
Spigot version: This server is running CraftBukkit version git-Spigot-4bd94dc-3a482bd (MC: 1.12.2) (Implementing API version 1.12.2-R0.1-SNAPSHOT)
ChatControlPro version:
ChatControl // Running 7.7.4
ChatControl // Created by kangarko _ 2013 - 2018

I have the following rule in my rules.txt:

````
# Warnings to player
match \bl+m+a+o+\b|\ba+s+s+\b
then deny
then log
then console bmwarn {player} [Auto Language Warning] You said: "{message}"|eventlogger &7[&eWARN&7] &9{player} &rjust said: &e{message}
````

Here is the pertinent info from my server log:
````
[Fri, 9. Mar 2018 17:35:46 EST INFO] [redacted player name] issued server command: /r lmao this wont create lag at all
[Fri, 9. Mar 2018 17:35:46 EST INFO] *--------- Rule match (command) for [redacted player name] --------- ID UNSET
[Fri, 9. Mar 2018 17:35:46 EST INFO] MATCH: \bl+m+a+o+\b|\ba+s+s+\b
[Fri, 9. Mar 2018 17:35:46 EST INFO] CATCH: /r lmao this wont create lag at all
[Fri, 9. Mar 2018 17:35:46 EST INFO] Original message cancelled.
[Fri, 9. Mar 2018 17:35:46 EST INFO] [redacted player name] has been warned by Creative for [Auto Language Warning] You said: "/r lmao this wont create lag at all"
````
**The player was issued a warning, but the message was still sent to the other player.**

This along with private messages being sent to random players or not sent at all is making me very worried with this 7.7 update.  Please advise ASAP.  I run a family-friendly server and our reputation is on the line if swear words get through our chat system.

**Comment by kangarko** [comment]
Hello,,

**1) Rules not filtering**
I just tested the rule by putting it to rules/chat.txt file and it was a successful match:
![a](https://i.imgur.com/W89uWgm.png)

Please answer the following to get it solved quickly:
- Are none of the rules working or just the one you created?
- Do you have any other plugin managing chat like EssentialsChat or DeluxeChat?

Please also try doing the followingquickly:
- Check the console for errors on startup (but I assumed you have done that).
- Try changing the listener priority of Checker as per [this](https://github.com/kangarko/ChatControl-Pro/wiki/Listener-Priorities) article to LOWEST, HIGHEST and MONITOR in that order. Restart after each change to check whether it will work. This is to eliminate any interference with another chat plugin.

**Comment by kangarko** [comment]
Hello,,

**1) Rules not filtering**
I just tested the rule by putting it to rules/chat.txt file and it was a successful match:
![a](https://i.imgur.com/W89uWgm.png)

Please answer the following to get it solved quickly:
- Are none of the rules working or just the one you created?
- Do you have any other plugin managing chat like EssentialsChat or DeluxeChat?

Please also try doing the following:
- Check the console for errors on startup (but I assumed you have done that).
- Try changing the listener priority of Checker as per [this](https://github.com/kangarko/ChatControl-Pro/wiki/Listener-Priorities) article to LOWEST, HIGHEST and MONITOR in that order. Restart after each change to check whether it will work. This is to eliminate any interference with another chat plugin.

**2) PM's being sent randomly**
Please provide steps on how to reproduce this issue. Where it most likely happens? Do you have BungeeCords ? Does it happen on /tell or /reply ? Anything that comes in mind ...

**Comment by c0wg0d** [comment]
I am not using any default rules.  I have not modified my rules in a very long time and they worked just fine in ChatControlPro 7.6.  And as you saw, my log also stated that the message was cancelled, but it was still sent to the other player.

I do not have any other plugins that modify chat, except for BanManagement which will not allow players to send messages if they are muted/tempmuted.

There are no errors on start up.

My priorities are set to:
```
Listener_Priority:
  Formatter: HIGH
  Checker: NORMAL
  Signs: HIGHEST
```

These have not changed in a long time and have always worked fine.

I'm not really sure what other information I can give to help you.

**Comment by kangarko** [comment]
Thank you for providing this information.

Okay, and what about your other rules? Is this the only one that's not working, more specifically, not being canceled?

I understand that even if you didn't make any specific changes to ChatControl or your server for some time. and since I was not able to reproduce on neither the public released version or on the next developed one, can you please try to experiment with the listener priorities anyways? Just change the Formatter back to NORMAL, and Checker to LOWEST or HIGHEST or MONITOR. From my point of view, if the rule is being caught and all operators function normally except for canceling, somewhere in the pipeline the event is un-cancelled again so the priorities may fix it.

**Comment by kangarko** [comment]
Just got the PM's issue fixed. I am pretty sure we can get this one solved as well. Please do the advice above and report ;)

**Comment by c0wg0d** [comment]
Upon further review, it appears that the issue is only with PM's.  The PM's seem like they are sent even if it breaks one of the chat filter rules.  If a message is sent in normal chat, it is indeed cancelled, but not if it's in a PM.

**Comment by kangarko** [comment]
Okay thanks for this info, going to rest asap.

**Comment by kangarko** [comment]
Works for me:

![a](https://i.imgur.com/2D041zs.png)

Do you use our private messaging system or have another plugin for that?

**Comment by c0wg0d** [comment]
I use ChatControl for PM's.  I will try changing the priority settings to see if it makes a difference.

**Comment by kangarko** [comment]
Closing this because I am clearing the entire Issues Channel _entirely_. By no means that is that I ignore you, just post below and I will get the same notification so I can answer quickly!

I hope that the priority solution will work. If not, post me your plugins' list just for a quick check.

Also, starting the next version, I added useful debug messages. Just set Debug to [rules] in the next version (right now it will break!) and you will get more information, perhaps very relevant to your case, in the console.
